addons/point_of_sale/models/pos_order.py

```diff
@@ -684,7 +685,7 @@ def _export_for_ui(self, order):
             'lines': [[0, 0, line] for line in order.lines.export_for_ui()],
             'statement_ids': [[0, 0, payment] for payment in order.payment_ids.export_for_ui()],
             'name': order.pos_reference,
-            'uid': order.pos_reference[6:],
+            'uid': re.search('([0-9]|-){12,}', order.pos_reference).group(0),
```

Comment:
can't we just take the 14 last character of the string?

Reply:
other fix : https://github.com/odoo/odoo/pull/71182

Reply:
No, because of the RTL languages 
For instance, in Hungarian:

Reply:
ok, why are you using 12? it seems that references are composed of 14 charachters

Reply:
Because the definition mentions 12 digits, so I wrote `>=12` in case the `-` are removed
https://github.com/odoo/odoo/blob/3e8d921318e86631b73c8c92c2e462fd0d761582/addons/point_of_sale/static/src/js/models.js#L2865-L2868
But I can change for 14 if you prefer ?

Reply:
yes maybe just set it to 14 and it is ok for me